be/src/exec/vectorized/json_parser.h

```diff
@@ -111,9 +175,25 @@ class ExpandedJsonArrayParserWithRoot : public JsonArrayParser {
 
 private:
     std::vector<SimpleJsonPath> _root_paths;
+
+    // data is parsed as an array, in which every document is an object and has an array under json root.
+
+    // _curr_row is the current document in array.
     simdjson::ondemand::object _curr_row;
+
+    // iterator context for array under json root.
     simdjson::ondemand::array _array;
     simdjson::ondemand::array_iterator _array_itr;
+
+    // Iterator (value, object, array, etc) in simdjson could be only parsed once.
+    // If we want to access iterator twice, a call of rewind/reset is needed.
+    // get_current would access many iterators, which are hard to reset and get the object.
+    // Hance, we keep the object returned in _curr and reset it when necessary.
+
+    // _curr is the object returned by get_current.
+    simdjson::ondemand::object _curr;
+    // _curr_ready denotes whether the _curr has been parsed.
+    bool _curr_ready = false;
```

Comment:
Can this field be represent by the `_curr` itself?

Reply:
It seems no good way to check whether the `_curr` is initiated.